io/jvm/src/main/scala/fs2/io/ioplatform.scala

```diff
@@ -144,4 +146,25 @@ private[fs2] trait ioplatform extends iojvmnative {
     }
   }
 
+  private lazy val vtExecutor: Option[ExecutionContext] = {
+    val javaVersion: Int =
+      System.getProperty("java.version").stripPrefix("1.").takeWhile(_.isDigit).toInt
+
+    // From JVM 21 on we can use virtual threads
+    if (javaVersion >= 21) {
+      val virtualThreadExecutor = classOf[Executors]
+        .getDeclaredMethod("newVirtualThreadPerTaskExecutor")
+        .invoke(null)
+        .asInstanceOf[ExecutorService]
+
+      ExecutionContext.fromExecutor(virtualThreadExecutor).some
+    } else {
+      None
+    }
+
+  }
+
+  def evalOnVirtualThreadIfAvailable[F[_]: Async, A](fa: F[A]): F[A] =
```

Comment:
Oh also, let's keep this private to FS2.
```suggestion
  private[io] def evalOnVirtualThreadIfAvailable[F[_]: Async, A](fa: F[A]): F[A] =
```